File: src/Web/trader-dashboard/src/types/order.ts
export type OrderStatus = 'Pending' | 'Processing' | 'Shipped' | 'Delivered' | 'Cancelled';

export interface OrderItem {
    id: string;
    productId: string;
    productName: string;
    quantity: number;
    unit: string;
    price: number;
    total: number;
}

export interface Order {
    id: string;
    orderNumber: string;
    customerName: string;
    customerEmail: string;
    customerPhone: string;
    shippingAddress: string;
    date: string;
    status: OrderStatus;
    totalAmount: number;
    items: OrderItem[];
    paymentStatus: 'Paid' | 'Unpaid' | 'COD';
}

File: src/Services/Order/KrishiEye.Services.Order.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace KrishiEye.Services.Order.Domain.Entities
{
    public class Order
    {
        public Guid Id { get; set; }
        public Guid BuyerId { get; set; }
        public Guid SellerId { get; set; }
        public decimal TotalAmount { get; set; }
        public OrderStatus Status { get; set; }
        public PaymentStatus PaymentStatus { get; set; }
        public ShippingAddress ShippingAddress { get; set; } = null!; // JSONB
        public Guid? TransportRequestId { get; set; }
        public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }

    public enum OrderStatus
    {
        Pending,
        Confirmed,
        Processing,
        Shipped,
        Delivered,
        Cancelled
    }

    public enum PaymentStatus
    {
        Pending,
        Paid,
        Failed
    }

    public class ShippingAddress
    {
        public string AddressLine { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string Division { get; set; } = string.Empty;
        public string PostalCode { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
    }
}
